app/main.py

```diff
@@ -2,0 +1,36 @@
+from __future__ import annotations
+
+
+class Animal:
+
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.name = name
+        self.hidden : bool = False
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    @classmethod
+    def animal_death(cls) -> None:
+        cls.alive = [animal for animal in cls.alive if animal.health > 0]
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, herbivore : Animal) -> None:
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                Animal.animal_death()
```

Comment:
The `animal_death` method is only called when a `Carnivore` bites a `Herbivore` and reduces its health to zero or below. This method should be called whenever an animal's health changes to ensure the `alive` list is updated correctly. Consider calling `animal_death` in the `__init__` method and whenever health is modified.